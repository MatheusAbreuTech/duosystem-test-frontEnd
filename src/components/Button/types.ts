import theme from '@/styles/theme'
import { ButtonHTMLAttributes } from 'react'

export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode
  formAction?: string
  color:
    | 'white'
    | 'black'
    | 'darkBlue'
    | 'lightBlue'
    | 'blueCyan'
    | 'red'
    | 'green'
    | 'transparent'
}

export interface ButtonStylesProps {
  theme: typeof theme
  color:
    | 'white'
    | 'black'
    | 'darkBlue'
    | 'lightBlue'
    | 'blueCyan'
    | 'red'
    | 'green'
    | 'transparent'
}
