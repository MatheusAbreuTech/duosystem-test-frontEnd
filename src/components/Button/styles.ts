import theme from '@/styles/theme'
import styled from 'styled-components'
import { ButtonStylesProps } from './types'

export const Button = styled.button<ButtonStylesProps>`
  background-color: ${({ color }) =>
    color === 'transparent' ? 'transparent' : theme.colors[color]};
  color: ${({ color }) =>
    color === 'transparent' ? theme.colors.darkBlue : theme.colors.white};
  border: none;
  border-radius: 0.5rem;
  padding: 1rem;
  max-height: 4.2rem;
  height: 100%;
  cursor: pointer;
  font-size: 1.6rem;
`
