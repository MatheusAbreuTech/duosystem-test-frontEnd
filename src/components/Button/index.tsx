import { ButtonHTMLAttributes, ReactNode } from 'react'
import * as S from './styles'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children?: ReactNode
  color:
    | 'white'
    | 'black'
    | 'darkBlue'
    | 'lightBlue'
    | 'blueCyan'
    | 'red'
    | 'transparent'
}

const Button = ({ children, color, ...props }: ButtonProps) => (
  <S.Button {...props} color={color}>
    {children ?? 'Label do bot√£o'}
  </S.Button>
)

export default Button
