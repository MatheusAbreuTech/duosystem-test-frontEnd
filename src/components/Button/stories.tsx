import { Meta, StoryFn, StoryObj } from '@storybook/react'
import Button from '.'
import { ButtonProps } from './types'

export default {
  title: 'Button',
  component: Button
} as Meta

export const Default: StoryObj = {}

const Template: StoryFn<ButtonProps> = (args) => <Button {...args} />
export const ButtonWithColorWhite = Template.bind({})
ButtonWithColorWhite.args = {
  color: 'white',
  label: 'bg-white'
}

export const ButtonWithColorBlack = Template.bind({})
ButtonWithColorBlack.args = {
  color: 'black',
  label: 'bg-black'
}

export const ButtonWithColorDarkBlue = Template.bind({})
ButtonWithColorDarkBlue.args = {
  color: 'darkBlue',
  label: 'bg-darkBlue'
}

export const ButtonWithColorLightBlue = Template.bind({})
ButtonWithColorLightBlue.args = {
  color: 'lightBlue',
  label: 'bg-lightBlue'
}

export const ButtonWithColorBlueCyan = Template.bind({})
ButtonWithColorBlueCyan.args = {
  color: 'blueCyan',
  label: 'bg-blueCyan'
}

export const ButtonWithColorRed = Template.bind({})
ButtonWithColorRed.args = {
  color: 'red',
  label: 'bg-red'
}

export const ButtonWithColorGreen = Template.bind({})
ButtonWithColorGreen.args = {
  color: 'green',
  label: 'bg-green'
}

export const ButtonWithColorTransparent = Template.bind({})
ButtonWithColorTransparent.args = {
  color: 'transparent',
  label: 'bg-transparent'
}
