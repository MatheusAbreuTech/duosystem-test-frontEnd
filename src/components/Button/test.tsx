import { render } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  test('should renders with default values', () => {
    const { getByText, container } = render(<Button color="white"></Button>)
    const defaultText = getByText('Label do botÃ£o')
    expect(defaultText).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should renders with provided children', () => {
    const buttonText = 'Click Me'
    const { getByText } = render(<Button color="white">{buttonText}</Button>)
    const renderedText = getByText(buttonText)
    expect(renderedText).toBeInTheDocument()
  })
})
