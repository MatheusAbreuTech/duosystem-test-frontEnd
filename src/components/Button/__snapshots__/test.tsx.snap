// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should renders with default values 1`] = `
.c0 {
  background-color: #ffffff;
  color: #ffffff;
  border: none;
  border-radius: 0.5rem;
  padding: 1rem;
  max-height: 4.2rem;
  height: 100%;
  cursor: pointer;
  font-size: 1.6rem;
}

<button
  class="c0"
  color="white"
>
  Label do bot√£o
</button>
`;
